artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/ActiveMQExceptionType.java

```diff
@@ -261,6 +261,12 @@ public ActiveMQException createException(String msg) {
       public ActiveMQException createException(String msg) {
          return new ActiveMQReplicationTimeooutException(msg);
       }
+   },
+   RESOURCE_LIMIT_REACHED(221) {
```

Comment:
What occurs with older clients that dont understand this new exception if this new exception is thrown and passed back to them

Reply:
For older client that don't understand, "ActiveMQException"(type code is GENERIC_EXCEPTION) would be decoded and thrown. There are two places where ActiveMQExceptionType.SESSION_CREATION_REJECTED is judged, one is ActiveMQSessionContext::recreateSession(), the other is BridgeImpl::connect(). For both of them, the logic is same in the case of GENERIC_EXCEPTION(old client) and RESOURCE_LIMIT_EXCEEDED(new client).

Reply:
People can use native core client. Not everyone uses jms.

Reply:
ActiveMQSessionContext::recreateSession() and BridgeImpl::connect() both are using native core client api.

Reply:
Point here just because theyre catching the base doesnt mean everyone is
 By changing the exception should try make it a extended of the current so people dont get any shock changes in their error handling logic.

Reply:
Like wise we may have remote clients using core, what occurs when server throws new exception but client is older, it wont know how to convert. As such need to handle this server side (most likely using something like the connection version to work out the client version and adjust for older)